internal/constants/scripts.go

```diff
@@ -196,3 +195,27 @@ if [ "$host_dir" ]
 fi
 echo "PreNetworkConfig End"
 `
+
+const PreNetworkConfigService = `
+[Unit]
+Description=Assisted static network config
+DefaultDependencies=no
+
+# We need to run after the kernel and udev have detected all the network interface devices,
+# otherwise the mechanism that we use to find the MAC addresses will not work and the configuration
+# will not be applied.
+After=systemd-udev-settle.service
```

Comment:
Note this service warns it's deprecated, because it's inherently racy.

Reply:
The alternative to this would be to explicitly wait for the devices that we expect, as you suggest in another comment, but that would greatly complicate the current script.